### ISSUE TITLE
Encoding abbreviations

### ISSUE THREAD

**Issue by danbalogh** [erc-dharma/project-documentation#146]
I'm moving this discussion from EGD comments to a new issue here.

History:

Arlo and Daniel discussed abbreviations and wanted to introduce encoding in this form: `<choice><abbr>mā</abbr><expan>māṣa</expan></choice>` (for _mā_ abbreviating _māṣa_). This is slightly redundant, but good TEI and simple and straightforward to encode even with complex abbreviations. However, it is not the standard EpiDoc solution, which would be `<expan><abbr>mā</abbr><ex>ṣa</ex></expan>`. Our problem with the EpiDoc standard is that it can get complex when the abbreviation includes characters from more than one part of a word (e.g. _kuvā_ abbreviating _kulyavāpa_, EpiDoc encoding  `<expan><abbr>ku</abbr><ex>lya</ex><abbr>vā</abbr><ex>pa</ex></expan>`), and even more so when the abbreviation includes characters that are not present in the resolution. (I can't think of an Indic example for that kind of abbreviation; English examples are _pp._ abbreviating _pages_ or _No_ abbreviating _number_; EpiDoc encoding `<expan><abbr>p</abbr><am>p.</am><ex>ages</ex></expan>`).

However, Axelle has pointed out that the EpiDoc schema requires that `<ex>` be present within `<expan>`, so we cannot proceed with the above naive suggestion unless we turn our backs on EpiDoc and use only the DHARMA schema in the future. I, Daniel, am very reluctant to do that. Even if DHARMA manages to set a new standard for diplomatic TEI encoding of Indic texts, if this standard contains elements that are contrary to EpiDoc standards (rather than a specification/restriction of EpiDoc), then it will be harder to get our standard accepted by the community and less hope for its continuation after our project.

As a way out, Axelle has suggested that we keep the `<choice>` method, but add `<ex>` as a sop to the EpiDoc schema. Thus, the above example would be encoded as `<choice><abbr>mā</abbr><expan>mā<ex>ṣa</ex></expan></choice>`. That method is endorsed by [one of the examples](https://tei-c.org/release/doc/tei-p5-doc/en/html/examples-expan.html#index-egXML-d39e34033) in the TEI documentation. However, this encoding is now no less complex than the standard EpiDoc practice. So at present my opinion is that we should continue using `<abbr>` to flag abbreviations without expanding them, and where an expansion is desired by the encoder, we stick to the EpiDoc method after all. Thus, I recommend

- for simple abbreviations: `<expan><abbr>mā</abbr><ex>ṣa</ex></expan>`, preferred over
  - `<choice><abbr>mā</abbr><expan>mā<ex>ṣa</ex></expan></choice>` (sop to EpiDoc) and
  - `<choice><abbr>mā</abbr><expan>māṣa</expan></choice>` (total rejection of EpiDoc)
- for abbreviations like _kuvā_:  `<expan><abbr>ku</abbr><ex>lya</ex><abbr>vā</abbr><ex>pa</ex></expan>`, preferred over
  - `<choice><abbr>kuvā</abbr><expan>ku<ex>lya</ex>vā<ex>pa</ex></expan></choice>` (sop to EpiDoc) and
  - `<choice><abbr>kuvā</abbr><expan>kulyavāpa</expan></choice>` (total rejection of EpiDoc)
- and for abbreviations like _pp._:  `<expan><abbr>p</abbr><am>p.</am><ex>ages</ex></expan>`, preferred over
  - `<choice><abbr>pp.</abbr><expan>p<ex>ages</ex></expan></choice>` (sop to EpiDoc) and
  - `<choice><abbr>pp.</abbr><expan>pages</expan></choice>` (total rejection of EpiDoc)
 
As you can see from the above examples, the "total rejection" way is the simplest to encode for complex abbreviations, but I don't think we want total rejection; and out of the two alternatives not involving rejection, the EpiDoc norm costs us no complexity in simple abbreviations and only a little extra complexity in complex abbreviations, but wins us clarity of encoding and compliance with an existing standard.

**Comment by ajaniak** [comment]
Sorry, I totally forgotten to answer this issue. 
I am ok with your suggestions Daniel.

**Comment by danbalogh** [comment]
Thanks, no problem. @arlogriffiths , please let me know if you agree too. This is not what we initially agreed on in the EGD comment, please read the details. I'll revise the EGD when I get the green light from you.

**Comment by arlogriffiths** [comment]
Sorry for letting this slip for so long. @danbalogh : Yes, we must remain EpiDoc-compliant. I completely agree with all your proposals made on 07/09/2021. Please revise the EGD accordingly, and close the relevant github issues and/or comment threads.

**Comment by danbalogh** [comment]
Thanks, Arlo.
I've written it up for the EGD (§7.3).
I've noted an error in my recommendations above: for the most complex case (like pp.), the EpiDoc guidelines actually say the `<am>` must be within the `<abbr>`. This makes good sense so I've kept it that way.

@ajaniak please make the necessary changes to the schema to allow the following forms of encoding. I guess this only calls for removing the present DHARMA constraints, since it's all EpiDoc.

- `<expan>` may contain one or more instances of `<abbr>` and `<ex>`
  - it is imo OK to retain the present EpiDoc rule that `<ex>` is mandatory within `<expan>`; perhaps, `<abbr>` should likewise be made mandatory since we don't foresee using `<expan>` without `<abbr>`
- `<abbr>` may appear on its own outside `<expan>`, as before
- `<abbr>` may contain `<am>` (in case someone ever needs it)
- `<ex>` may have the attribute `@cert="low"` (as per the EpiDoc guidelines)

I guess it will be best to discuss display with Arlo and Manu. My suggestion is the following:
- `<abbr>` on its own should be displayed as text, without any special feature
- `<expan>` should be displayed with an attached tooltip and a different font colour
  - the displayed text should consist of the contents of `<abbr>` and `<am>` (if present) within the `<expan>` (but should not include the contents of `<ex>`)
  - the tooltip should consist of the contents of  the `<abbr>` and `<ex>` elements within the `<expan>` (but should not include the contents of `<am>`, if present)
- instead of a tooltip, we might want to consider showing everything in the text (in which case perhaps the contents of `<ex>` could be shown in the same way as `<supplied reason="omitted">` and the contents of `<am>` could be shown in the same way as `<surplus>`
